Sources/core/objects/Tables.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "Movable.h"
+
+class Tables: public IMovable
+{
+public:
+    DECLARE_SAVED(Tables, IMovable);
+    DECLARE_GET_TYPE_ITEM(Tables);
+    Tables(size_t id);
+    virtual void AttackBy(IdPtr<Item> item) override;
+    virtual void Delete() override;
+    void CheckSurroundings(bool table);
+    void UpdateSprite();
+    void SetTable(Dir directon, bool table);
+    virtual void AfterWorldCreation() override;
+protected:
+    std::string KV_SAVEBLE(material_);
+private:
+    bool KV_SAVEBLE(up_);
+    bool KV_SAVEBLE(down_);
+    bool KV_SAVEBLE(left_);
+    bool KV_SAVEBLE(right_);
+    bool KV_SAVEBLE(upright_);
+    bool KV_SAVEBLE(downright_);
+    bool KV_SAVEBLE(upleft_);
+    bool KV_SAVEBLE(downleft_);
+};
```

Comment:
It is not needed to keep nearest tables in the table. Atmos pipes do such things because they interact with nearest pipes each tick or so. Tables are not processing, so it is fine just scan nearest area on a table creation and destruction.